plugins/duckduckgo.py

```diff
@@ -1,185 +1,165 @@
-            result = result[index] if len(result) > index else None
-        if not result: continue
-
-        if result.text: response = result.text
-        if result.text and hasattr(result, 'url') and urls:
-            if result.url: response += ' (%s)' % result.url
-        if response: break
-
-    # if there still isn't anything, try to get the first web result
-    if not response and web_fallback:
-        if ddg.redirect.url:
-            response = ddg.redirect.url
-
-    # final fallback
-    if not response:
-        response = 'Sorry, no results.'
-
-    return response
+        css_class = ''
+        href = ''
+        for k, v in attr:
+            if k == 'class':
+                css_class = v
+            elif k == 'href':
+                href = v
+
+        if tag == 'a' and css_class == 'result__snippet' and href != '':
+            try:
+                self._url = get_real_url(f'https:{href}')
+                self._in_result = True
+            except ValueError:
+                pass
+
+        # ddg puts <b> tags around our search terms.
+        # let us make them bold.
+        elif tag == 'b' and self._in_result:
+            self._snippet += "\x02"
+
+    def handle_endtag(self, tag):
+        """Append a result or do nothing."""
+        if tag == 'a' and self._in_result:
+            self.result = DdgResult(self._url, self._snippet)
+            self._in_result = False
+
+        # Terminate the bold text.
+        elif tag == 'b' and self._in_result:
+            self._snippet += "\x02"
+
+    def handle_data(self, data):
+        """Append snippet text."""
+        if self._in_result:
+            self._snippet += data
+
+
+def get_answer(q: str) -> str:
+    """
+    Search ddg for a given query.
 
+    Args:
+        q: The user supplied query.
 
-@hook.command
-def ddg(inp):
-    """.ddg <search> -- search something on duckduckgo"""
-    return get_zci(inp)
+    Returns:
+        Result suitable for emitting to IRC.
+    """
+    with requests.get(SEARCH_ENGINE,
+                      params={'q': q},
+                      headers={'User-Agent': 'Taigabot/20220618'},
+                      allow_redirects=True,
+                      stream=True,
+                      timeout=15) as stream:
+        parser = DdgQueryParser()
+        # Up to 128KiB read.
+        for (frag, _) in zip(stream.iter_content(chunk_size=4096), range(10)):
+            if not frag:
+                break
+            parser.feed(frag.decode('utf8'))
+            # We only want one result.
+            if parser.result:
+                break
+
+        if parser.result:
+            url, snippet = parser.result
+            if len(snippet) > LEN_LIMIT:
+                snippet = f'{snippet[0:LEN_LIMIT]}...'
+            return f'''{snippet} - {url}'''
+        else:
+            return 'No result.'
+
+
+@hook.command('ddg', autohelp=False)
+def main(inp: str, say: Callable):
```

Comment:
can u make the command `ddg` please? i think people wont use `.main`

(yea the function name is the command)

Reply:
i realize you put the command in the decorator... either way this will add a `.main` command